lib/styles.js

```diff
@@ -340,4 +342,133 @@ module.exports = function (expect) {
             this.sp().commentBlock(pen);
         }
     });
+
+    expect.addStyle('diffFragment', function (text, baseStyle, specialCharStyle, markUpSpecialCharacters) {
+        if (markUpSpecialCharacters) {
+            text.split(specialCharRegExp).forEach(function (part) {
+                if (specialCharRegExp.test(part)) {
+                    this[specialCharStyle || baseStyle](utils.escapeChar(part));
+                } else {
+                    this[baseStyle](part);
+                }
+            }, this);
+        } else {
+            this[baseStyle](text);
+        }
+    });
+
+    expect.addStyle('stringDiff', function (actual, expected, options) {
+        options = options || {};
+        var type = options.type || 'WordsWithSpace';
+        var diffLines = [];
+        var lastPart;
+        stringDiff.diffLines(actual, expected).forEach(function (part) {
```

Comment:
I think the the diff formatting codes looks a bit weird, maybe it would also help to just start over. Make a utils.stringDiff that produces a data structure that contains lines of tokens. Then hand that output to a magicpen style that would format it the way we want. Then it should also be possible to get rid of `prependLinesWith'.